### PR TITLE
FE Admin UI update - New field to accept this value

### PR DESCRIPTION
**Pull request by gilluminate**
Closes [PROD-1709](https://ethyca.atlassian.net/browse/PROD-1709)

### Description Of Changes

Adds new field to the translation form for Trigger Link Label. This allows users to set the language of the link that opens the Fides modal on their site.

### Code Changes

* [x] adds new `trigger_link_label` to the type definitions of `ExperienceTranslationCreate` and `ExperienceTranslation` schema.
* [x] adds new `trigger_link_label` to form inclusions
* [x] adds new field to the Admin UI translation form (WIP. Not hooked up to BE yet) 

### Steps to Confirm

* [ ] Open an experience in Admin UI
* [ ] Select a language from the sidebar
* [ ] Scroll to the bottom and add a new label to the Trigger Link Label field

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* [x] Issue Requirements are Met

[PROD-1709]: https://ethyca.atlassian.net/browse/PROD-1709?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: clients/admin-ui/src/types/api/models/ExperienceTranslation.ts [hunk 1]
```diff
@@ -57,4 +57,8 @@ export type ExperienceTranslation = {
    * Overall description - used for banner as well if applicable.  HTML descriptions are supported so links can be included.
    */
   description?: string;
+  /**
+   * Custom link/button trigger label
+   */
+  trigger_link_label?: string;
```
Comment (eastandwestwind) [inline clients/admin-ui/src/types/api/models/ExperienceTranslation.ts:63]: can we rename this across the board to `modal_link_label`? This is clearer that the link opens the modal, kind of analogous to how we name the `privacy_policy_link_label`, where this link opens the privacy policy.

File: clients/fides-js/src/lib/i18n/locales/index.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ import type { Locale, Messages, Language } from "..";
  * NOTE: This process isn't automatic. To add a new static locale, follow these steps:
  * 1) Add the static import of the new ./{locale}/messages.json file
  * 2) Add the locale to the LOCALES object below
+ * 3) Add the locale to the LOCALE_LANGUAGE_MAP object below
```
Comment (eastandwestwind) [inline clients/fides-js/src/lib/i18n/locales/index.ts:10]: Mega nit-pick, but I see the `LOCALE_LANGUAGE_MAP` but now I'm curious where the `LOCALES` obj line 9 is referring to? Has this been renamed to something else?

File: clients/admin-ui/src/features/privacy-experience/PrivacyExperienceTranslationForm.tsx [hunk 3]
```diff
@@ -296,6 +296,15 @@ const PrivacyExperienceTranslationForm = ({
           variant="stacked"
         />
       ) : null}
+      {!!formConfig.trigger_link_label?.included && (
```
Comment (eastandwestwind) [inline clients/admin-ui/src/features/privacy-experience/PrivacyExperienceTranslationForm.tsx:299]: `.included` should always be a boolean. Can we follow the convention in this file and just use:

```suggestion
      {formConfig.trigger_link_label?.included && (
```

**Review by eastandwestwind — CHANGES_REQUESTED** [review]
Couple small nits, mostly want to agree on naming of the new var. Thanks!

File: clients/admin-ui/src/types/api/models/ExperienceTranslation.ts (hunk 1 above)
Comment (gilluminate) [reply clients/admin-ui/src/types/api/models/ExperienceTranslation.ts:63]: love it.

File: clients/fides-js/src/lib/i18n/locales/index.ts (hunk 2 above)
Comment (gilluminate) [reply clients/fides-js/src/lib/i18n/locales/index.ts:10]: Good catch! I'm assuming it got renamed to `STATIC_MESSAGES` in this doc. Looks like I also need to add this to the tcf version of this file.

File: clients/admin-ui/src/features/privacy-experience/PrivacyExperienceTranslationForm.tsx (hunk 3 above)
Comment (gilluminate) [reply clients/admin-ui/src/features/privacy-experience/PrivacyExperienceTranslationForm.tsx:299]: with the `?` there's a potential for it to be `undefined` instead of `boolean`, which is why I add the coercion. The result is the same though, so I'm not opposed.

Comment (gilluminate) [reply clients/admin-ui/src/features/privacy-experience/PrivacyExperienceTranslationForm.tsx]: FWIW, I believe the use of the ternary operators above are an anti-pattern. Using `&&` in those places is a much cleaner approach. Thoughts?

Comment (eastandwestwind) [reply clients/admin-ui/src/features/privacy-experience/PrivacyExperienceTranslationForm.tsx]: mostly I just want the file to be consistent, so feel free to refactor! I'm all for using && instead of ternary :)

File: clients/fides-js/src/components/ConsentButtons.tsx [hunk 4]
```diff
@@ -59,6 +59,7 @@ export const ConsentButtons = ({
                 data-testid={`fides-i18n-option-${lang.locale}`}
                 onClick={() => handleLocaleSelect(lang.locale)}
                 isActive={currentLocale === lang.locale}
+                title={lang.label_en}
```
Comment (eastandwestwind) [inline clients/fides-js/src/components/ConsentButtons.tsx:62]: why is this line added?

Comment (gilluminate) [reply clients/fides-js/src/components/ConsentButtons.tsx:62]: I was thinking it might be nice to have the english version of the language translations as a tooltip when you hover over each language but forgot to add that before. Overkill?

Comment (eastandwestwind) [reply clients/fides-js/src/components/ConsentButtons.tsx:62]: ohhhh I see haha. We'll just want to confirm with @mfbrown

Comment (gilluminate) [reply clients/fides-js/src/components/ConsentButtons.tsx:62]: I'll pull it out of this PR and add it to the new story he just created for me that's more related. I'll ask him about it there :) https://ethyca.atlassian.net/browse/PROD-1908

**Review by eastandwestwind — APPROVED** [review]
